performant-software/neatline-spa
# Display interactive record layers on map

### What does this PR do?
- Adds functionality to load records from the API once an exhibit has been opened
- Displays coverage (as geoJSON) and WMS (as WMS overlay) data from records as layers on the map
- Adds a layer switcher control, to which WMS layers are added as toggle-able overlays
- Adds interactivity for geoJSON points: when hovered over, an info panel will show the record's title; when clicked, the info panel will open with the record's title and body
- Modifies coverage layers to display as circle markers rather than the generic marker image, with stroke and fill styles responsive to mouse interaction

### What issues does it address?
- Closes #2 
- Closes #3 

### How to test
- Visit https://neatline-3-staging.herokuapp.com/s/public-site/neatline to see the top-level Neatline view for a public Omeka S site
- Click "Public View" on the first exhibit in the table to open its public map view
- The map should show one or more circle markers corresponding to records. Moving your cursor over a circle marker should cause its title to appear in a small info panel in the top left of the map. Moving the cursor away should cause the panel to disappear.
- Click on a circle marker. This should cause its title and body text to appear in the info panel, which will stay open when the cursor moves away. Clicking on another record's marker should populate the info panel with its information. The info panel should close when its 'x' icon is clicked, when an empty spot on the map is clicked, or when the currently selected record is clicked again.
- A circle marker should appear slightly more opaque when it is selected (when the info panel is displaying its record). Circle markers should appear noticeably more opaque when the cursor is passing above them.
- Hovering over the layer menu icon in the upper right of the map should show a divided list with OpenStreetMap (the base layer) in the first part and any WMS overlays from records in the second part. WMS layers should be named with their record's title, and their visibility should be toggle-able with a checkbox in the menu.
